### ISSUE TITLE
Ban return foo in Array.forEach / (Iterable Object).forEach

### ISSUE THREAD

**Issue by hfhchan** [eslint/eslint#10849]
**Please describe what the rule should do:**
There should not be any `return foo` statement in Array.forEach / (Iterable Object).forEach loops.  Only `return;` and `return void 0;` should be allowed to early exit the loop.

This is the exact opposite of "array-callback-return".

**What category of rule is this? (place an "X" next to just one item)**
[ ] Enforces code style
[ X ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**
```js
const output = rows.forEach(row => {
  let item = {};
  // do something with item
  return item; // should throw an error
});
```

**Why should this rule be included in ESLint (instead of a plugin)?**
When refactoring to avoid the create-array-loop-mutate-push-array anti-pattern, it is easy to forget to change the forEach call into a map call if the mutate logic is very long. 

Enforcing no return value for .forEach complements the rule to enforce return values in .map/.filter/.reduce.

**Comment by kaicataldo** [comment]
This seems valuable to me. I wonder if this would make sense to include as an option in the `array-callback-return` rule? We could then make it the default in a future major version.
